assets/src/components/detourListPage.tsx

```diff
@@ -1,21 +1,52 @@
-import React, { useCallback, useState } from "react"
+import React, { useCallback, useEffect, useState } from "react"
 import { DetoursTable } from "./detoursTable"
 import userInTestGroup, { TestGroups } from "../userInTestGroup"
 import { Button } from "react-bootstrap"
 import { PlusSquare } from "../helpers/bsIcons"
 import { DetourModal } from "./detours/detourModal"
-import { fetchDetours } from "../api"
+import { fetchDetour, fetchDetours } from "../api"
 import { useApiCall } from "../hooks/useApiCall"
 import { isOk } from "../util/result"
+import { isValidSnapshot } from "../util/isValidSnapshot"
+import { createDetourMachine } from "../models/createDetourMachine"
+import { Snapshot } from "xstate"
 
 export const DetourListPage = () => {
   const [showDetourModal, setShowDetourModal] = useState(false)
+  const [detourId, setDetourId] = useState<number | undefined>()
+  const [stateOfDetourModal, setStateOfDetourModal] =
+    useState<Snapshot<unknown> | null>()
 
   const { result } = useApiCall({
     apiCall: useCallback(async () => fetchDetours(), []),
   })
   const detours = result && isOk(result) && result.ok
 
+  const { result: detourResult } = useApiCall({
+    apiCall: useCallback(
+      async () => (detourId ? fetchDetour(detourId) : undefined),
+      [detourId]
+    ),
+  })
+  const detour =
+    detourResult && isOk(detourResult) ? detourResult.ok : undefined
+
+  useEffect(() => {
+    const state = isValidSnapshot(createDetourMachine, detour?.state)
+    if (isOk(state)) setStateOfDetourModal(state.ok)
+  }, [detour])
+
```

Comment:
you could also do this for `detours` _if_ desired, but that's less relevant because it doesn't require 2+n render loops

```
  const { result: detours } = useApiCall({
    apiCall: useCallback(async () => {
      const detoursResult = await fetchDetours()

      if (isErr(detoursResult)) { return undefined }

      return detoursResult.ok
    }, []),
  })
```

Reply:
Beeeaauuutiful, yeah I love this